src/protocol/mod.rs

```diff
@@ -94,24 +105,34 @@ impl Step {
 #[cfg(test)]
 mod parse_step {
     use super::*;
+    use pretty_assertions::assert_eq;
     use test_utils::assert_error;
-    use yaml_rust::Yaml;
+    use yaml_rust::YamlMarked;
 
     fn test_parse_step(yaml: &str) -> R<Step> {
-        let yaml = YamlLoader::load_from_str(yaml)?;
+        let yaml = YamlLoader::load_from_str_with_markers(yaml)?;
         assert_eq!(yaml.len(), 1);
         let yaml = &yaml[0];
         Step::parse(yaml)
     }
 
+    fn step_with_marker(command_matcher: CommandMatcher, marker: Marker) -> Step {
+        let mut step = Step::new(command_matcher);
+        step.marker = Some(marker);
+        step
+    }
+
     #[test]
     fn parses_strings_to_steps() -> R<()> {
         assert_eq!(
             test_parse_step(r#""foo""#)?,
-            Step::new(CommandMatcher::ExactMatch(Command {
-                executable: PathBuf::from("foo"),
-                arguments: vec![],
-            })),
+            step_with_marker(
+                CommandMatcher::ExactMatch(Command {
+                    executable: PathBuf::from("foo"),
+                    arguments: vec![],
+                }),
+                Marker { line: 1, col: 0 }
```

Comment:
since the marker is a default value, what do you think about making the function above `step_with_default_marker` and not specifying the marker explicitly in the tests?

Reply:
I was kinda torn on this. The tests make an equality comparison between `Step` values, and if the marker is not specified correctly in the expected value then the test will fail. We could change the tests to compare specific properties within a `Step` instead of the whole structure - for example we could compare just the command and arguments. In that case calling `Step::new` would suffice instead of `step_with_marker`.

Honestly I'm trying to suppress my instinct to put multiple assertions in a test (to check equality for command and arguments) in favor of the idea of testing just one thing in each test. I suppose I could put the commands and arguments into tuples for comparisons to technically get one assertion. I might make that change.